include/swift/Option/FrontendOptions.td

```diff
@@ -26,6 +26,8 @@ def primary_file : Separate<["-"], "primary-file">,
 
 def filelist : Separate<["-"], "filelist">,
   HelpText<"Specify source inputs in a file rather than on the command line">;
+def primary_filelist : Separate<["-"], "primary-filelist">,
+  HelpText<"Specify primary inputs in a file in addition to the command line">;
```

Comment:
I maintain that this should be "rather than" not "in addition to". Allowing primary files to be specified in both places just makes it harder to tell what a particular command line is doing.

Reply:
Have added the check, a test, and a diagnostic for that.